File: examples/csharp/Controllers/HomeController.cs
using Microsoft.AspNetCore.Mvc;

namespace RecurrenteCheckout.Controllers
{
    public class HomeController : Controller
    {
        public IActionResult Index()
        {
            return View();
        }

        public IActionResult Success(string checkoutId)
        {
            ViewBag.CheckoutId = checkoutId;
            return View();
        }

        public IActionResult Failure()
        {
            return View();
        }
    }
}

File: examples/java/src/main/java/com/recurrente/checkout/controller/HomeController.java
package com.recurrente.checkout.controller;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;

@Controller
public class HomeController {

    @GetMapping("/")
    public String index() {
        return "index";
    }

    @GetMapping("/success")
    public String success(@RequestParam(value = "checkoutId", required = false) String checkoutId, Model model) {
        model.addAttribute("checkoutId", checkoutId);
        return "success";
    }

    @GetMapping("/failure")
    public String failure() {
        return "failure";
    }
}
